From 5291bb9614c295af4deaefa7c9f448891e72d2ba Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Karel=20Ko=C4=8D=C3=AD?= <cynerd@email.cz>
Date: Mon, 19 Oct 2020 11:34:03 +0200
Subject: [PATCH] ath10k-ct-firmware: add conflicts and better provides

This expands packages to define not only provides but also conflicts.
These packages provides same files so they should specify conflicts.

Second expansion is that *-ct-htt firmwares can also provide *-ct
variant as those make it also compatible.
---
 package/firmware/ath10k-ct-firmware/Makefile | 24 +++++++++++++++-----
 1 file changed, 18 insertions(+), 6 deletions(-)

diff --git a/package/firmware/ath10k-ct-firmware/Makefile b/package/firmware/ath10k-ct-firmware/Makefile
index 12fb031..20514fa 100644
--- a/package/firmware/ath10k-ct-firmware/Makefile
+++ b/package/firmware/ath10k-ct-firmware/Makefile
@@ -191,13 +191,15 @@ $(Package/ath10k-ct-firmware-default)
   SECTION:=firmware
   CATEGORY:=Firmware
   PROVIDES:=ath10k-firmware-qca988x
+  CONFLICTS:=ath10k-firmware-qca988x
 endef
 define Package/ath10k-firmware-qca988x-ct-htt
 $(Package/ath10k-ct-firmware-default)
   TITLE:=ath10k CT 10.1 htt-mgt fw for QCA988x
   SECTION:=firmware
   CATEGORY:=Firmware
-  PROVIDES:=ath10k-firmware-qca988x
+  PROVIDES:=ath10k-firmware-qca988x ath10k-firmware-qca988x-ct
+  CONFLICTS:=ath10k-firmware-qca988x ath10k-firmware-qca988x-ct
   DEPENDS:=+!PACKAGE_kmod-ath10k-ct-smallbuffers:kmod-ath10k-ct
 endef
 
@@ -207,13 +209,15 @@ $(Package/ath10k-ct-firmware-default)
   SECTION:=firmware
   CATEGORY:=Firmware
   PROVIDES:=ath10k-firmware-qca9887
+  CONFLICTS:=ath10k-firmware-qca9887
 endef
 define Package/ath10k-firmware-qca9887-ct-htt
 $(Package/ath10k-ct-firmware-default)
   TITLE:=ath10k CT 10.1 htt-mgt fw for QCA9887
   SECTION:=firmware
   CATEGORY:=Firmware
-  PROVIDES:=ath10k-firmware-qca9887
+  PROVIDES:=ath10k-firmware-qca9887 ath10k-firmware-qca9887-ct
+  CONFLICTS:=ath10k-firmware-qca9887 ath10k-firmware-qca9887-ct
   DEPENDS:=+!PACKAGE_kmod-ath10k-ct-smallbuffers:kmod-ath10k-ct
 endef
 
@@ -223,13 +227,15 @@ $(Package/ath10k-ct-firmware-default)
   SECTION:=firmware
   CATEGORY:=Firmware
   PROVIDES:=ath10k-firmware-qca99x0
+  CONFLICTS:=ath10k-firmware-qca99x0
 endef
 define Package/ath10k-firmware-qca99x0-ct-htt
 $(Package/ath10k-ct-firmware-default)
   TITLE:=ath10k CT 10.4 htt-mgt fw for QCA99x0
   SECTION:=firmware
   CATEGORY:=Firmware
-  PROVIDES:=ath10k-firmware-qca99x0
+  PROVIDES:=ath10k-firmware-qca99x0 ath10k-firmware-qca99x0-ct
+  CONFLICTS:=ath10k-firmware-qca99x0 ath10k-firmware-qca99x0-ct
   DEPENDS:=+!PACKAGE_kmod-ath10k-ct-smallbuffers:kmod-ath10k-ct
 endef
 
@@ -239,13 +245,15 @@ $(Package/ath10k-ct-firmware-default)
   SECTION:=firmware
   CATEGORY:=Firmware
   PROVIDES:=ath10k-firmware-qca9984
+  CONFLICTS:=ath10k-firmware-qca9984
 endef
 define Package/ath10k-firmware-qca9984-ct-htt
 $(Package/ath10k-ct-firmware-default)
   TITLE:=ath10k CT 10.4 htt-mgt fw for QCA9984
   SECTION:=firmware
   CATEGORY:=Firmware
-  PROVIDES:=ath10k-firmware-qca9984
+  PROVIDES:=ath10k-firmware-qca9984 ath10k-firmware-qca9984-ct
+  CONFLICTS:=ath10k-firmware-qca9984 ath10k-firmware-qca9984-ct
   DEPENDS:=+!PACKAGE_kmod-ath10k-ct-smallbuffers:kmod-ath10k-ct
 endef
 
@@ -255,13 +263,15 @@ $(Package/ath10k-ct-firmware-default)
   SECTION:=firmware
   CATEGORY:=Firmware
   PROVIDES:=ath10k-firmware-qca4019
+  CONFLICTS:=ath10k-firmware-qca4019
 endef
 define Package/ath10k-firmware-qca4019-ct-htt
 $(Package/ath10k-ct-firmware-default)
   TITLE:=ath10k CT 10.4 htt-mgt for QCA4018/9
   SECTION:=firmware
   CATEGORY:=Firmware
-  PROVIDES:=ath10k-firmware-qca4019
+  PROVIDES:=ath10k-firmware-qca4019 ath10k-firmware-qca4019-ct
+  CONFLICTS:=ath10k-firmware-qca4019 ath10k-firmware-qca4019-ct
   DEPENDS:=+!PACKAGE_kmod-ath10k-ct-smallbuffers:kmod-ath10k-ct
 endef
 
@@ -271,13 +281,15 @@ $(Package/ath10k-ct-firmware-default)
   SECTION:=firmware
   CATEGORY:=Firmware
   PROVIDES:=ath10k-firmware-qca9888
+  CONFLICTS:=ath10k-firmware-qca9888
 endef
 define Package/ath10k-firmware-qca9888-ct-htt
 $(Package/ath10k-ct-firmware-default)
   TITLE:=ath10k CT 10.4 htt-mgt fw for QCA9886/8
   SECTION:=firmware
   CATEGORY:=Firmware
-  PROVIDES:=ath10k-firmware-qca9888
+  PROVIDES:=ath10k-firmware-qca9888 ath10k-firmware-qca9888-ct
+  CONFLICTS:=ath10k-firmware-qca9888 ath10k-firmware-qca9888-ct
   DEPENDS:=+!PACKAGE_kmod-ath10k-ct-smallbuffers:kmod-ath10k-ct
 endef
 
-- 
2.28.0

